service/schedule/scheduleRunner.go

```diff
@@ -270,8 +264,20 @@ func ScheduledLabelFilter() string {
 	return strings.Join(filter, ", ")
 }
 
-func (s *scheduleRunner) waitForLock(name string, backend string, jobs []observe.JobName) {
+func (s *scheduleRunner) waitForLock(name string, backend string, jobs []observe.JobName, jobType string) {
+	queue := queueCount.WithLabelValues(s.schedule.GetNamespace(), jobType)
+
 	s.Logger.Infof("%v for repo %v is queued waiting for jobs %v to finish", name, backend, jobs)
+	queue.Inc()
 	s.observer.GetLocker().WaitForRun(backend, jobs)
 	s.Logger.Infof("All blocking jobs on %v for %v are now finished", backend, name)
+	queue.Dec()
```

Comment:
This is a good place to make use of `defer`.

Reply:
Nice catch! That should lower the risk of deadlocks even further!